packages/cardano-services/src/TxSubmit/openApi.json

```
@@ -0,0 +1,75 @@
+{
+  "openapi": "3.0.0",
+  "info": {
+    "title": "Tx Submit",
+    "license": {
+      "name": "Apache 2.0",
+      "url": "http://www.apache.org/licenses/LICENSE-2.0.html"
+    },
+    "version": "1.0.0"
+  },
+  "paths": {
+    "/tx-submit/submit": {
+      "post": {
+        "summary": "submits given transaction",
+        "operationId": "submitTx",
+        "requestBody": {
+          "content": {
+            "application/json": {
+              "schema": {
+                "$ref": "#/components/schemas/txsubmit_submit_body"
+              }
+            }
+          },
+          "required": true
+        },
+        "responses": {
+          "200": {
+            "description": "success operation",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "object"
+                }
+              }
+            }
+          },
+          "400": {
+            "description": "invalid request",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "type": "object"
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "components": {
+    "schemas": {
+      "TipifiedValue": {
+        "required": ["value"],
+        "type": "object",
+        "properties": {
+          "value": {
+            "type": "string"
+          }
+        }
+      },
+      "txsubmit_submit_body": {
+        "type": "object",
+        "properties": {
+          "args": {
+            "type": "array",
+            "items": {
+              "type": "object"
```

UInt8Array gets serialized to hex string via `toSerializableObject`
```suggestion
              "type": "string"
```

---

If I do so, tests will fail. I think that it expects an object with this format `{ __type: 'Buffer', value: string }`

---

Ah yes, you're right - that's what it is. Perhaps specify this type in openApi.json too then?

---

For some reason if one field starts with double underscore the other fields get ignore , looks like a bug on open api validator. I let the fields of the object as optional for now. Ive tried with this format and works perfectly `{ _type: string, value: string }` but I dont like much the idea to adapt the code to the library.

---

Changing field name is still an option at this point. Maybe we could come up with something less conflict-prone than `_type` :thinking: This field is not supposed to be used in code directly, only by toSerializableObject and fromSerializableObject, so we can even use special characters if we want.

---

Maybe something like `_objType` will be less conflictibe

---

Ok, I see what is going on. It wasnt the naming being used. It is actually the bodyParser middleware that maps the request when it invokes `fromSerializableObject`. So, in this case we have a Uint8Array object and the initially schema is the one which works. This leads the openApi doc a bit confusing.